rfc/website-revamp.md

```diff
@@ -0,0 +1,148 @@
+* Provide a clear and compelling answer to anyone asking, “Why Spinnaker for Continuous Delivery?”
+* Add useful documentation features, e.g. versioned documentation and a better in-page table of contents
+* Improve operator, end-user, and contributor experience by providing more navigable documentation as well as a search bar on the spinnaker.io landing page.
+* Improve content organization to facilitate easy contribution.
+* Improve local build time and adopt the new way GitHub pages uses themes
+
+### Non-Goals
+
+* Not a complete reorganization of the documentation set. What will happen is the current documentation related tabs will be condensed under a single “Docs” tab in the top navigation.
+* Not replatforming or rethemeing .io. The cost/benefit is not there to do this at this time.
+
+## Motivation and Rationale
+
+Spinnaker.io has operated with one basic design look/feel since March 22, 2017. Given the current momentum and size of the project, it is time for a refresh. The Docs SIG feels it’s time to evolve .io beyond a documentation site and into a project hub, providing a presence for Spinnaker that is clearly segmented for different user personas.
+
+We need to provide context to attract new companies and ICs as they encounter Spinnaker. This aligns with Travis Tomsu’s efforts to add more context and persuasive oomph to READMEs across the project. The window for Spinnaker inevitability is threatened by a crowded marketplace and will close; this project will boost Spinnaker’s profile and chances for inevitability.
+
+From a technical standpoint, the way GitHub pages manages themes has changed to remote themes, and our implementation is outdated. Relying on the Minimal Mistakes theme-maintainer’s repository results in less theme-related maintenance for the Docs SIG and capture of theme feature improvements.
+
+## Timeline
+* Achieve buy-in from stakeholders by March.
+* Finalize Statement of Work and share with community in March.
+* Obtain funding for a contractor from the Continuous Delivery Foundation after Statement of Work has been finalized.
+* Finalize SIG-validated high-level design in April.
+* Sitemap
+* Accept contractor bids, including billing rate and time estimate, in April.
+* Identify and hire webdev/design contractor in April-May.
+* Work begins in May, validated with Docs SIG continuously.
+* Work completed by July 1.
+
+## Design
+
+### Spinnaker Personas
+* Individual Contributor: I am a developer who contributes code to the core Spinnaker project, or maintains a fork or custom extensions of Spinnaker for my company.
+* Operator: I manage a Spinnaker installation for my company.
+* App Developer: I am on an application team and use Spinnaker to manage my continuous delivery workflow as an end-user.
+* Non-technical stakeholder/Executive: I am evaluating Spinnaker as a means to achieve some business objective.
+
+### Top Nav Menu
```

Comment:
There's an effort as part of Spinnaker as a Platform (plugins, etc) that will auto-generate reference docs and samples from code. My strategy here is that we'll be able to generate and export Jekyll content that could be consumed by the spinnaker.io repo.

Would this class of documentation just go under "Docs" under a sub-category of "Reference" or something like that? This is a totally new way of providing docs into the website, so I'm just curious on thoughts. Since the docs will be served from outside locations, it seems to me that it should be somewhat segregated in the navigation so that we don't accidentally stomp on docs written against the main repo.

Reply:
It would go in Docs > Reference. We are planning to have that also be the location of the current Halyard command reference, which is also autogenerated from my understanding. Parallel to this website project, @aimeeu is also working on a docs contributor guide where we can note all the auto generated docs.

Reply:
Yes - working on contributor docs and a SIG Docs release handbook based on my Kubernetes SIG Docs experience. Right now both are low priority, though,  as I work on plugin and operator docs. Thanks for everyone's patience!

Reply:
Great news! Thanks!